### PR TITLE
Fsf c

### PR DESCRIPTION
**Pull request by igorljubuncic**
New flow for C - including a full dosbox example.

### CODE REVIEW

File: webapp/first_snap/content/c/package.yaml [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    - text: In general, snaps cannot see the root filesystem on end user systems. This prevents conflict with other applications and increases security. However, applications still need some location to act as the root filesystem. They would also benefit from common libraries (e.g. libc) being in this root filesystem rather than bundled into each application.
+    - text: The base keyword specifies a special kind of snap that provides a minimal set of libraries common to most applications. It will be mounted as the root filesystem for your application.
+    - code: |
+        base: core18
+    - text: The <code>core18</code> base is recommended.
+  parts:
+    - text: |
+        Parts define what sources are needed to assemble your app. Parts can be anything: programs, libraries, or other needed assets. In this case we have one: the DOSBox source release tarball. In other cases these can point to local directories, remote git repositories or other revision control systems.
+    - text: |
+        Before building the part, the dependencies listed as <code>build-packages</code> are installed.
```
Comment (evandandrea) [inline webapp/first_snap/content/c/package.yaml]: This sounds like they're going to be installed on my system, but they're actually going to be installed inside the VM that Snapcraft uses when creating the snap.

File: webapp/first_snap/content/c/package.yaml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+            build-packages:
+              - g++
+              - make
+              - libsdl1.2-dev
+              - libpng-dev
+              - libsdl-net1.2-dev
+              - libsdl-sound1.2-dev
+              - libasound2-dev
+    - text: |
+        The <code>stage-packages</code> section specifies the list of libraries required by the DOSBox snap at runtime.
```
Comment (evandandrea) [inline webapp/first_snap/content/c/package.yaml]: Something like:
>The <code>stage-packages</code> section specifies the list of libraries required by the DOSBox snap at runtime. These will be included in the snap, ensuring that every system can run dosbox regardless of what libraries are already installed.

This also looks disconnected to the rest of the parts definition. I'd consider showing the whole parts section once with the explanatory text beneath it, like we do on the other guides.

File: webapp/first_snap/content/c/package.yaml [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+              - libvorbisenc2
+              - libwrap0
+              - libx11-6
+              - libxau6
+              - libxcb1
+              - libxdmcp6
+              - libxext6
+              - zlib1g
+    - text: |
+            The <code>autotools</code> plugin uses the standard tools - configure and make to build the part. The command <code>snapcraft help autotools</code> gives more information on how this plugin works.
```
Comment (evandandrea) [inline webapp/first_snap/content/c/package.yaml]: This feels odd here. How about making it the first thing said after the parts section is displayed?

**Comment by evandandrea** [comment]
Confirmed that the DOSBox example builds. Committing to https://github.com/snapcraft-docs/dosbox now.

**Comment by igorljubuncic** [comment]
Made changes as requested.

**Comment by bartaz** [comment]
@igorljubuncic Could you rebase your PR on top of fresh master? This should hopefully fix the issues with test coverage dropping.

**Pull request merged**